src/commands/init.js

```diff
@@ -6,6 +6,8 @@ const path = require('path')
 const fs = require('fs-extra')
 const { installDeps } = require('../util')
 const defaultAPMName = require('../helpers/default-apm')
+import { checkProjectExists } from '../lib/check-project-exists';
+import { prepareTemplate } from '../lib/prepare-template';
```

Comment:
As discussed, we're introducing a `lib` folder here that would have all the actual functions etc. that can be used to compose CLI commands.

Also notice that I'm using `import` syntax here. This doesn't look very nice as the project so far seems to use commonjs syntax. I'm happy to change that to commonjs, however, I'd personally vote for rather slowly moving to ECMAScript module syntax as this is an official standard (by now).

If you agree with that, we can decide whether we change those imports in one go, or progressively as we go along (also here my personally vote: better progressively and keep shipping at the cost of having some temporary inconsistencies)

Reply:
Looks good. The `lib` directory may end up having a ton of files if we extract all individual tasks. I propose we create a directory for every command (the ones you have created now would go under `lib/init/`).

If we do that, how would you feel about having an index file per command in the lib dir and export all the different tasks?

```
import { prepareTemplate, checkProjectExists } from '../lib/init'
```

For the import/exports, I agree on changing as we go.

Reply:
> I propose we create a directory for every command (the ones you have created now would go under lib/init/).

We can certainly do that, just keep in mind that there might be lib functions that will be used across tasks, or maybe even other lib functions. However, lib functions that exists only for a dedicated command, then definitely go in corresponding sub folders.

> If we do that, how would you feel about having an index file per command in the lib dir and export all the different tasks?

Sure, we can do that. No strong feelings about that. :)